### ISSUE TITLE
Installing on Cygwin

### ISSUE THREAD

**Issue by abers** [DTVD/rainbowstream#179]
I am having an issue installing rainbowstream due to python on Cygwin using cp1252 as default encoder: 
>     Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 20, in <module>
      File "E:\cygwin\tmp\pip-build-68cao31c\rainbowstream\setup.py", line 31, in <module>
        long_description=open("./README.rst", "r").read(),
      File "c:\python\python35-32\lib\encodings\cp1252.py", line 23, in decode
        return codecs.charmap_decode(input,self.errors,decoding_table)[0]
    UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 2898: character maps to <undefined>

I know I need to change it to utf-8 but can't find any easy explanations for how to do this - any help with this much appreciated.

**Comment by urbster1** [comment]
this does install on cygwin, but you probably need more packages than you think. try doing a full cygwin install (if need be, you can leave out gnome, KDE, and the window managers). i had this issue too but after installing more packages it worked. sorry i can't be of more help and tell you which one actually fixed it though.

**Comment by abers** [comment]
Thanks for the reply. Tried a full cygwin install on both PC and laptop the latter still can't install it and provides the same error and the former can install it but provides following error when trying to run:

> The system cannot find the path specified.
Traceback (most recent call last):
  File "c:\python27\lib\runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "c:\python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\Scripts\rainbowstream.exe\__main__.py", line 5, in <module>
  File "c:\python27\lib\site-packages\rainbowstream\rainbow.py", line 22, in <module>
    from .draw import *
  File "c:\python27\lib\site-packages\rainbowstream\draw.py", line 15, in <module>
    from .c_image import *
  File "c:\python27\lib\site-packages\rainbowstream\c_image.py", line 25, in <module>
    rgb2short = call_c()
  File "c:\python27\lib\site-packages\rainbowstream\c_image.py", line 21, in call_c
    image_c = ctypes.cdll.LoadLibrary(library)
  File "c:\python27\lib\ctypes\__init__.py", line 443, in LoadLibrary
    return self._dlltype(name)
  File "c:\python27\lib\ctypes\__init__.py", line 365, in __init__
    self._handle = _dlopen(self._name, mode)
WindowsError: [Error 126] The specified module could not be found

**Comment by urbster1** [comment]
do you have python3 and if so have you tried doing 'pip3 install rainbowstream' ?

**Comment by abers** [comment]
On PC I unstalled rainbowstream and tried pip3 instead. Unfortunately, it doesn't install providing similar error message to trying to install it via pip & pip3 on the laptop:

> Collecting rainbowstream
  Using cached rainbowstream-1.3.3.tar.gz
    Complete output from command python setup.py egg_info:
    cp: failed to access 'E:Programs\cygwin64\home\Alasdair\.rainbow_config.json': Bad address
    chmod: cannot access 'E:Programs\cygwin64\home\Alasdair\.rainbow_config.json': Bad address
    Traceback (most recent call last):
      File "<string>", line 20, in <module>
      File "E:\Programs\cygwin64\tmp\pip-build-73h0laue\rainbowstream\setup.py", line 31, in <module>
        long_description=open("./README.rst", "r").read(),
      File "c:\users\alasdair\appdata\local\programs\python\python35-32\lib\encodings\cp1252.py", line 23, in decode
        return codecs.charmap_decode(input,self.errors,decoding_table)[0]
    UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 2898: character maps to <undefined>
Command "python setup.py egg_info" failed with error code 1 in E:\Programs\cygwin64\tmp\pip-build-73h0laue\rainbowstream

**Comment by DTVD** [comment]
I'm sorry RS is not officially support Window environment, even inside Cygwin. I'm not sure if people made it worked or not, but i highly recommend using a Linux/Mac environment so you can save all the headache

**Comment by abers** [comment]
No problem, I thought that was the case but wasn't sure if someone had unofficially figured out a way to get it to work. From what I've found searching about there are folk like urbster1 where its worked after installing other packages, but not sure which exact ones, and then folk like me where it just doesn't seen to work, even after a full install. Managed to install it easy enough on my raspberry pi, and can always ssh into it from PC and laptop.
